src/main/java/core/dao/car/CarDaoImpl.java

```diff
@@ -0,0 +1,48 @@
+package core.dao.car;
+
+import core.lib.Dao;
+import core.model.Car;
+import core.storage.Storage;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class CarDaoImpl implements CarDao {
+    @Override
+    public Car add(Car car) {
+        Storage.add(car);
+        return car;
+    }
+    
+    @Override
+    public Optional<Car> get(Long id) {
+        return Optional.ofNullable(Storage.carStorage.get(id));
+    }
+    
+    @Override
+    public List<Car> getAll() {
+        return new ArrayList<>(Storage.carStorage.values());
+    }
+    
+    @Override
+    public Car update(Car car) {
+        Car oldCar = Storage.carStorage.get(car.getId());
+        Storage.carStorage.put(car.getId(), car);
+        return oldCar;
+    }
+    
+    @Override
+    public boolean delete(Long id) {
+        if (Storage.carStorage.containsKey(id)) {
+            Storage.carStorage.remove(id);
+            return true;
+        }
+        return false;
+    }
+    
+    @Override
+    public boolean delete(Car car) {
+        return delete(car.getId());
+    }
+}
```

Comment:
![Untitled](https://user-images.githubusercontent.com/59320919/104361472-29481580-551b-11eb-9bdf-67fa106330ff.png)
you should implement this methods too, I saw you did this in carServiceImpl, but I not sure in this

Reply:
well, DAO layer supposed to interact with DB, and give us Car instances and don't alter them. So to add or remove drivers as part of a car is altering the object which is the task of service. Thus in DAO I can CRUD DB, but with Service I perform some functionality over cars and update the car in DB with DAO. So I don't think these methods have to be in CarDao